src/js/jsx/tools/SuperselectVectorMaskOverlay.jsx

```diff
@@ -0,0 +1,346 @@
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, 
+ * and/or sell copies of the Software, and to permit persons to whom the 
+ * Software is furnished to do so, subject to the following conditions:
+ *  
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *  
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, 
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER 
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING 
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
+ * DEALINGS IN THE SOFTWARE.
+ * 
+ */
+
+define(function (require, exports, module) {
+    "use strict";
+
+    var React = require("react"),
+        Fluxxor = require("fluxxor"),
+        FluxMixin = Fluxxor.FluxMixin(React),
+        StoreWatchMixin = Fluxxor.StoreWatchMixin,
+        d3 = require("d3");
+
+    var UI = require("adapter").ps.ui;
+
+    var system = require("js/util/system"),
+        keyUtil = require("js/util/key");
+
+    var SuperselectVectorMaskOverlay = React.createClass({
+        mixins: [FluxMixin, StoreWatchMixin("tool", "document", "application", "ui", "modifier")],
+
+        /**
+         * Owner group for the vector mask HUD elements
+         *
+         * @type {SVGElement}
+         */
+        _vectorMaskHudGroup: null,
+
+        /**
+         * UI Scale for drawing strokes visible at all zoom levels
+         *
+         * @type {number}
+         */
+        _scale: null,
+
+        getStateFromFlux: function () {
+            var flux = this.getFlux(),
+                applicationStore = flux.store("application"),
+                toolStore = flux.store("tool"),
+                modalState = toolStore.getModalToolState(),
+                currentDocument = applicationStore.getCurrentDocument(),
+                vectorMaskMode = toolStore.getVectorMode();
+
+            return {
+                document: currentDocument,
+                modalState: modalState,
+                vectorMaskMode: vectorMaskMode
+            };
+        },
+
+        componentDidUpdate: function (prevProps, prevState) {
+            if (this.state.vectorMaskMode !== prevState.vectorMaskMode) {
+                this.drawOverlay();
+            }
+        },
+
+        /**
+         * Calls all helper functions to draw super-select overlay
+         * Cleans it first
+         */
+        drawOverlay: function () {
+            if (!this.isMounted()) {
+                return;
+            }
+
+            var currentDocument = this.state.document,
+                svg = d3.select(React.findDOMNode(this));
+
+            svg.selectAll(".superselect-bounds").remove();
```

Comment:
the blinking went away, i think We should leave this since we use that class at times when the HUD is active